src/neptune/metadata_containers/run.py

```diff
@@ -451,6 +464,35 @@ def _write_initial_attributes(self):
             # upload default sources ONLY if creating a new run
             upload_source_code(source_files=self._source_files, run=self)
 
+    def _track_dependencies(self) -> None:
+        if self._dependencies == "infer":
+            # using pipreqs here
+            proc = subprocess.Popen(["pipreqs", "--print", "."], stdout=subprocess.PIPE)
+            proc.wait()
+
+            dependencies_str = proc.communicate()[0].decode(encoding="utf-8")
+
+            if not dependencies_str:
+                from neptune.internal.utils.logger import logger
+
+                logger.error("Could not generate requirements. Call to 'pipreqs' returned an empty string.")
+                return
+
+            self["source_code/files/requirements"].upload(File.from_content(dependencies_str))
```

Comment:
I think it could be problematic if we're going to upload some source files under `source_code/files`. Did you check that?